crates/rustc_codegen_spirv/src/linker/test.rs

```diff
@@ -80,67 +118,62 @@ fn assemble_and_link(binaries: &[&[u8]]) -> Result<Module, PrettyString> {
         let matches = rustc_driver::handle_options(&["".to_string(), "x.rs".to_string()]).unwrap();
         let sopts = rustc_session::config::build_session_options(&matches);
 
-        rustc_interface::util::run_in_thread_pool_with_globals(
-            sopts.edition,
-            sopts.unstable_opts.threads,
-            || {
-                let mut sess = rustc_session::build_session(
-                    sopts,
-                    None,
-                    None,
-                    Registry::new(&[]),
-                    rustc_session::DiagnosticOutput::Default,
-                    Default::default(),
+        run_in_thread_pool_with_globals(sopts.edition, sopts.unstable_opts.threads, || {
```

Comment:
```suggestion
        rustc_span::create_session_globals_then(sopts.edition, || {
```
Does this work? If so we wouldn't need `run_in_thread_pool_with_globals` (I would've avoided it if I had noticed it got privated, I should've double-checked...).

Reply:
Lol yeah much simpler